src/openrct2-ui/UiContext.Linux.cpp

```diff
@@ -160,7 +160,20 @@ namespace OpenRCT2 { namespace Ui
                 std::string output;
                 if (Execute(cmd, &output) == 0)
                 {
-                    result = output;
+                    // The default file extension is taken from the **first** available filter, since
+                    // we cannot obtain it from zenity's output. This means that the FileDialogDesc::Filters
+                    // array must be carefully populated, at least the first element.
+                    std::string pattern = desc.Filters[0].Pattern;
+                    std::string defaultExtension = pattern.substr(pattern.find_last_of('.'));
+                    int dotPosition = output.size() - extension.size();
+                    if (output.substr(dotPosition, defaultExtension.size()).compare(defaultExtension) == 0)
```

Comment:
This looks a whole lot better. But can you put `Add the default extension if no extension is specified.` above line 169? (It may sound inconsistent with my earlier feedback, but finding a balance between giving enough information and stating the obvious is not always easy.)

Reply:
Thank you very much.
The last commit adds that comment and also fixes the build errors (forgot to rename "extension" -> "defaultExtension" in line 168